fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayBanner.kt

```diff
@@ -7,65 +7,190 @@ package org.mozilla.fenix.tabstray
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
+import mozilla.components.ui.tabcounter.TabCounter
+import org.mozilla.fenix.R
 import org.mozilla.fenix.compose.annotation.LightDarkPreview
-import org.mozilla.fenix.compose.button.PrimaryButton
 import org.mozilla.fenix.theme.FirefoxTheme
+import java.text.NumberFormat
 
 /**
  * Top-level UI for displaying the banner in [TabsTray].
  *
  * @param isInMultiSelectMode Whether the tab list is in multi-select mode.
+ * @param selectedPage The active [Page] of the Tabs Tray.
+ * @param normalTabCount The total amount of normal browsing tabs currently open.
  * @param onTabPageIndicatorClicked Invoked when the user clicks on a tab page indicator.
  */
 @Composable
 fun TabsTrayBanner(
     isInMultiSelectMode: Boolean,
+    selectedPage: Page,
+    normalTabCount: Int,
     onTabPageIndicatorClicked: (Page) -> Unit,
 ) {
     if (isInMultiSelectMode) {
         MultiSelectBanner()
     } else {
         SingleSelectBanner(
             onTabPageIndicatorClicked = onTabPageIndicatorClicked,
+            selectedPage = selectedPage,
+            normalTabCount = normalTabCount,
         )
     }
 }
 
+@Suppress("LongMethod")
 @Composable
 private fun SingleSelectBanner(
+    selectedPage: Page,
+    normalTabCount: Int,
     onTabPageIndicatorClicked: (Page) -> Unit,
 ) {
-    Row(
-        modifier = Modifier
-            .fillMaxWidth()
-            .height(80.dp)
-            .padding(all = 16.dp),
-        horizontalArrangement = Arrangement.spacedBy(16.dp),
-    ) {
-        Box(modifier = Modifier.weight(1.0f)) {
-            PrimaryButton(text = "Normal tabs") {
-                onTabPageIndicatorClicked(Page.NormalTabs)
-            }
-        }
+    val normalTabCountText: String
+    val normalTabCountTextModifier: Modifier
+    if (normalTabCount > TabCounter.MAX_VISIBLE_TABS) {
+        normalTabCountText = TabCounter.SO_MANY_TABS_OPEN
+        normalTabCountTextModifier = Modifier.padding(bottom = 1.dp)
+    } else {
+        normalTabCountText = NumberFormat.getInstance().format(normalTabCount.toLong())
+        normalTabCountTextModifier = Modifier
+    }
+    val normalTabsContentDescription = if (normalTabCount == 1) {
+        stringResource(id = R.string.mozac_tab_counter_open_tab_tray_single)
+    } else {
+        String.format(
+            stringResource(id = R.string.mozac_tab_counter_open_tab_tray_plural),
+            normalTabCount.toString(),
+        )
+    }
```

Comment:
Would it be worth creating a separate composable for the normal tabs icon? Just feels kinda busy to have all these declarations in this (already fairly large) function when they're only used once.

Reply:
There's undoubtedly some value. I'll make a private one in the file.